lib/src/mock_document_snapshot.dart

```diff
@@ -4,15 +4,14 @@ import 'package:cloud_firestore_mocks/src/util.dart';
 
 import 'util.dart';
 
-class MockDocumentSnapshot implements DocumentSnapshot {
+class MockDocumentSnapshot<T extends Object?> implements DocumentSnapshot<T> {
```

Comment:
Having `?` here is strange.

Reply:
Agreed. I am just following the new API.

https://pub.dev/documentation/cloud_firestore/2.1.0/cloud_firestore/DocumentSnapshot-class.html